index.js

```diff
@@ -90,14 +91,32 @@ class Sqlite extends Base {
     ], cb)
   }
 
-  upsert (data, cb) {
-    const res = this._buildUpsertQuery(data)
-    this.db.run(res.query, res.data, function (err) {
-      if (err) return cb(err)
-      cb(null, { lastID: this.lastID })
+  async runAsync (sql, params = []) {
+    return new Promise((resolve, reject) => {
+      return this.db.run(sql, params, function (err, res) { // passing an arrow function won't work
+        if (err) return reject(err)
+        return resolve(this)
+      })
     })
   }
 
+  async allAsync (sql, params, ...rest) {
+    return promisify(this.db.all.bind(this.db))(sql, params, ...rest)
```

Comment:
hmm this would be performance issue as we would end up always promisifying these, does it work to have this on start e.g. like this:
```js
...
      next => {
        this.db.configure('busyTimeout', this.opts.busyTimeout || 1000)
        this.db.on('error', err => {
          console.error('ERR_DB_SQLITE', err)
        })
        this._maybeRunSqlAtStart(next)
      },
      (next) => {
        this.allAsync = promisify(this.db.all.bind(this.db))
        this.getAsync = promisify(this.db.get.bind(this.db))
        this.execAsync = promisify(this.db.exec.bind(this.db))
        next()
      }
```